staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -72,6 +72,59 @@ func init() {
 	dbMetricsMonitors = make(map[string]struct{})
 }
 
+type activeETCD3Client struct {
+	client  *clientv3.Client
+	destroy DestroyFunc
+	refs    int
+}
+
+type etcd3ClientCache struct {
+	mx     sync.Mutex
+	active map[string]*activeETCD3Client
+}
+
+func newETCD3ClientCache() *etcd3ClientCache {
+	return &etcd3ClientCache{active: make(map[string]*activeETCD3Client)}
+}
+
+// For returns a compacted etcd v3 client for the supplied transport config. One
+// client and compactor are started for each transport. Callers must call the
+// returned destroy function once they are done with the client.
+func (e *etcd3ClientCache) For(tc storagebackend.TransportConfig, compactionInterval time.Duration) (*clientv3.Client, DestroyFunc, error) {
+	e.mx.Lock()
+	defer e.mx.Unlock()
+
+	k := fmt.Sprintf("%v", tc) // gives: {[server1 server2] keyFile certFile caFile}
```

Comment:
something like https://github.com/kubernetes/kubernetes/blob/d046a58de435a07cf5d0d2d0879c3b21dbb52cca/staging/src/k8s.io/client-go/transport/cache.go#L60-L66

Reply:
Can you elaborate on what the advantage to doing this would be? In this case I don't think we have any sensitive data to redact.

(FWIW I copied this from the code that was used to create a key per transport for compaction earlier).

Reply:
it is just for having control of the the key format

Reply:
maybe it is me being paranoid

Reply:
> it is just for having control of the the key format

Ah, I see your point. I'm on the fence here and leaning toward leaving it as is, since these keys are only in memory anyhow. I don't feel very strongly though so let me know if you really want it changed.